### PR TITLE
Reduce duplicated library interactives

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/177968044

[#177968044]

### CODE REVIEW

File: app/models/library_interactive.rb [hunk 1]
```diff
@@ -81,11 +81,17 @@ def export
   end
 
   def generate_export_hash
-    Digest::SHA1.hexdigest(export().to_json)
+    export = export()
+    export.except("authoring_guidance", "description", "name")
```
Comment (scytacki) [inline app/models/library_interactive.rb:85]: this won't actually remove these properties from export it just returns a new value that has them removed if you use the bang version instead: `except!(...)` that will remove them from the export variable.

**Review by scytacki — CHANGES_REQUESTED** [review]
I'd recommend adding some spec tests of the new generate_export_hash function. They ought to expose the error in the code I found.

For example you could make a test that builds two library interactives where the only difference is their description. The value of generate_export_hash should match. And if the two library interactives have different base_urls then the generate_export_hash should not match.

File: app/models/library_interactive.rb (hunk 1 above)
Comment (scytacki) [reply app/models/library_interactive.rb:85]: For reference I tested this in the rails console with:
```
irb(main):001:0> example = {hello: "hi", bye: "bye"}
=> {:hello=>"hi", :bye=>"bye"}
irb(main):003:0> examplej = example.as_json
=> {"hello"=>"hi", "bye"=>"bye"}
irb(main):004:0> examplej.except("hello")
=> {"bye"=>"bye"}
irb(main):005:0> examplej
=> {"hello"=>"hi", "bye"=>"bye"}
irb(main):006:0> examplej.except!("hello")
=> {"bye"=>"bye"}
irb(main):007:0> examplej
=> {"bye"=>"bye"}
```

**Comment by emcelroy** [comment]
@scytacki Thanks. Are the tests that I've added enough, do you think?

**Review by scytacki — APPROVED** [review]
Looks good to me.

**Comment by scytacki** [comment]
@emcelroy can you add the next LARA release version label to the associated story.

**Comment by emcelroy** [comment]
@scytacki Sure. I've added the latest version label.

**Pull request merged**